src/client/app/effects/snippet.ts

```diff
@@ -34,6 +34,7 @@ export class SnippetEffects {
                 .map((snippet: ISnippet) => ({ snippet, mode }))
                 .filter(({ snippet }) => !(snippet == null))
                 .mergeMap(({ snippet, mode }) => this._massageSnippet(snippet, mode, isViewMode))
+                .mergeMap(actions => (actions))
```

Comment:
Did you try:
 .mergeMap(({ snippet, mode }) => Observable
            .fromPromise(this._massageSnippet(snippet, mode, isViewMode)))